Tooling/ScriptGenerator/Services/GeneratorService.cs

```diff
@@ -0,0 +1,151 @@
+    /// <param name="scriptMutations">A list of mutations to be performed on a script.</param>
+    /// <exception cref="ArgumentNullException">
+    ///     Thrown if any of the constructor parameters are null.
+    /// </exception>
+    public GeneratorService(
+        IDirectory directory,
+        IFile file,
+        IPath path,
+        IConsoleService consoleService,
+        IMethodExtractorService methodExtractorService,
+        IRelativePathResolverService pathResolver,
+        IScriptTemplateService scriptTemplateService,
+        IStringMutation[] scriptMutations)
+    {
+        EnsureThat.ParamIsNotNull(directory);
+        EnsureThat.ParamIsNotNull(file);
+        EnsureThat.ParamIsNotNull(path);
+        EnsureThat.ParamIsNotNull(consoleService);
+        EnsureThat.ParamIsNotNull(methodExtractorService);
+        EnsureThat.ParamIsNotNull(pathResolver);
+        EnsureThat.ParamIsNotNull(scriptTemplateService);
+        EnsureThat.ParamIsNotNull(scriptMutations);
+
+        this.directory = directory;
+        this.file = file;
+        this.path = path;
+        this.consoleService = consoleService;
+        this.methodExtractorService = methodExtractorService;
+        this.pathResolver = pathResolver;
+        this.scriptTemplateService = scriptTemplateService;
+        this.scriptMutations = scriptMutations;
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="ArgumentNullException">
+    ///     Occurs if any of the parameters are null or empty.
+    /// </exception>
+    /// <exception cref="DirectoryNotFoundException">
+    ///     Occurs if the given <paramref name="destDir"/> does not exist.
+    /// </exception>
+    /// <exception cref="FileNotFoundException">
+    ///     Occurs if the file at the given <paramref name="srcFilePath"/> does not exist.
+    /// </exception>
+    public void GenerateScript(string srcFilePath, string destDir, string destFileName)
+    {
+        this.consoleService.WriteLine($"Original Source File Path: {srcFilePath}");
+        this.consoleService.WriteLine($"Original Destination Directory Path: {destDir}");
+        this.consoleService.WriteLine($"Original Destination File Name: {destFileName}");
+
+        if (string.IsNullOrEmpty(srcFilePath))
+        {
+            throw new ArgumentNullException(nameof(srcFilePath), "The parameter must not be null or empty.");
+        }
+
+        srcFilePath = this.pathResolver.Resolve(srcFilePath);
+
+        this.consoleService.WriteLine($"Resolving Source File Path To: {srcFilePath}");
+
+        if (string.IsNullOrEmpty(destDir))
+        {
+            throw new ArgumentNullException(nameof(destDir), "The parameter must not be null or empty.");
+        }
+
+        destDir = this.pathResolver.Resolve(destDir);
+
+        if (destDir.StartsWith($".{this.path.DirectorySeparatorChar}") ||
+            destDir.StartsWith($".{this.path.AltDirectorySeparatorChar}"))
+        {
+            destDir = destDir[2..];
```

Comment:
This is highlighted in red and I don't know how to fix. Refactor if needbe.
```suggestion
            destDir = destDir[**_2..]_**;
```